ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,42 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
+      }
+    }
+    return name
+  }
+
+  async putCourseName (userLoginId: string, courseId: number, newName: string): Promise<string | null> {
```

Comment:
I think something useful to consider is incorporating a request id.
For example, `putCourseName `in `api.service.ts` can accept a [guid ](https://www.npmjs.com/package/guid-typescript)and `putCourseName` in `api.controller.ts` can generate one and forward it on to be used in logging.  You can use this info when trying to isolate a single request 'thread' when there is a lot of activity going on that would otherwise make it difficult to tell what is going on.

You can do stuff like 
`logger.log('putCourseName[ '+ requestId +'] userLoginId: ' + userLoginId +' courseId: ' + ... )`

Honestly I have no idea if there is a design pattern or best practice around this, I just started doing it in my own projects because it seemed useful.  Maybe there is a better way to implement it.

Reply:
I like this idea, it does seem useful. I don't really like adding more parameters that will need to be present in all API service methods, though maybe it's unavoidable if you want to trace a connection between the method calls. @chrisrrowland, are you alright with me looking into this separately? I can create an issue based on your comment. In the meantime, I'm going to look into whether I can add any more `debug` level logging that might be useful.

Reply:
See https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/121/commits/866b34e1dd01989f7d085a5adb3a59067ef549d2 for some debug logging.

Reply:
That's fine

Reply:
Great, see https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/128